packages/nextjs/services/api/batches/index.ts

```diff
@@ -0,0 +1,96 @@
+import { SortingState } from "@tanstack/react-table";
+import { Batch, BatchWithCounts } from "~~/services/database/repositories/batches";
+
+export const getSortedBatches = async (start: number, size: number, sorting: SortingState, filter?: string) => {
+  const response = await fetch(
+    `/api/batches/sorted?start=${start}&size=${size}&sorting=${JSON.stringify(sorting)}&filter=${filter || ""}`,
+  );
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch sorted batches: ${response.status} ${response.statusText}`);
+  }
+
+  const batchesData = (await response.json()) as {
+    data: BatchWithCounts[];
+    meta: {
+      totalRowCount: number;
+    };
+  };
+
+  return batchesData;
+};
+
+export async function createBatch({
+  address,
+  signature,
+  name,
+  startDate,
+  status,
+  contractAddress,
+  telegramLink,
+  bgSubdomain,
+}: {
+  address: string;
+  signature: string;
+  name: string;
+  startDate: string;
+  status: string;
+  contractAddress?: string;
+  telegramLink: string;
+  bgSubdomain: string;
+}) {
+  const response = await fetch("/api/batches/create", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ address, signature, name, startDate, status, contractAddress, telegramLink, bgSubdomain }),
+  });
+
+  const data = await response.json();
+
+  if (!response.ok) {
+    throw new Error(data.error || "Batch creation failed");
+  }
+
+  return data.batch as Batch;
+}
+
+export async function editBatch(
```

Comment:
Shall we call this `updateBatch` ?  similar to other functions like `updateLocation` in `services/api/users.ts`

Reply:
I thought about it, but named `edit` for distinction between client modals components, so modalContent will not have the same name as editModal
<img width="1089" alt="image" src="https://github.com/user-attachments/assets/34e23085-c56a-4c5f-b6d1-4163c7b2db68" />

Reply:
Umm lol yeah I see, I am like 60% on `updateBatch` in this file since like it a services or util irrespective of where it's called. So maybe naming it based on client code isn't that much worth it? Lol I mean it's being used at only place that too in client code so no strong opinion and we could keep it as it 🙌